client/router_create_test.go

```diff
@@ -338,14 +344,23 @@ func TestRouterCreateDefaults(t *testing.T) {
 				rolesFound = append(rolesFound, role.Name)
 			},
 		})
+
+		expectedClusterRoles := sets.NewString(c.clusterRolesExpected...)
 		clusterRoleInformer := clusterRoleInformerFactory.Rbac().V1().ClusterRoles().Informer()
 		clusterRoleInformer.AddEventHandler(&cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				clusterRole := obj.(*rbacv1.ClusterRole)
 				if strings.HasPrefix(clusterRole.Name, "skupper") {
-					clusterRolesFound = append(clusterRolesFound, clusterRole.Name)
-					for _, p := range clusterRole.Rules {
-						clusterRolesResourcesFound = clusterRolesResourcesFound.Insert(p.Resources...)
+					if isCluster && !expectedClusterRoles.Has(clusterRole.Name) {
```

Comment:
I am wondering if this is the appropriate thing to do, or maybe change the comparison done
below with `cmp.Diff` to something that simply ensures all expectedClusterRoles have been found?

Reply:
I just tried it (see updated branch).  The problem is with the `clusterRolesResourcesFound` test; the rules are added into a set, with no reference back to the clusterRole that included them.  Without special treatment, it fails as follows:

```
 === RUN   TestRouterCreateDefaults
     router_create_test.go:461: TestRouterCreateDefaults test two cluster roles mismatch (-want +got):
           []string(Inverse(Sort, []string{
           	"skupper-service-controller-basic",
         + 	"skupper-service-controller-extended",
           }))
     router_create_test.go:462: Checking specifically for expected cluster roles
     router_create_test.go:471: TestRouterCreateDefaults test two cluster roles policy resources mismatch (-want +got):
           sets.String(
         - 	{"namespaces": {}, "nodes": {}, "skupperclusterpolicies": {}},
         + 	{"namespaces": {}, "nodes": {}, "pods": {}, "skupperclusterpolicies": {}},
```

ie, the clusterRole test works fine, ignoring the additional `skupper-service-controller-extended` role, but the `policy resources` (rules) test fails with a rule that comes from the ignored role.

I could refactor `clusterRolesResourcesFound`, so that it becomes a map from clusterRole name to set of rules...  But then I'd have to either specify ruleset per role on the test table, or check that each expected rule is present on at least one role and the other way around (every rule from a role is present on the  `clusterRolesResourcesFound`).

That would make the test a bit better, actually, but also more complex; is it worth it?

The original approach seemed cleaner to me, though.  The end result is the same, in any case: unexpected additional roles get logged and ignored, when running in a cluster.

Reply:
@fgiorgetti, as we talked, I reverted the changes back to the original approach.  It passed CI and it is ready for review again